XiEditor/AppDelegate.swift

```diff
@@ -228,12 +228,34 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
         }
     }
 
+    // Called when using the 'Open Recent' menu. If we return false, the item
+    // is removed from that menu.
+    func application(_ sender: NSApplication, openFile filename: String) -> Bool {
+        if FileManager.default.fileExists(atPath: filename) {
+            documentController.openDocumentAsync(atUrl: URL(fileURLWithPath: filename))
+            return true
+        } else {
+            return false
+        }
+    }
+
     // MARK: - XiClient protocol
 
-    func update(viewIdentifier: String, update: [String: AnyObject], rev: UInt64?) {
-        let document = documentForViewIdentifier(viewIdentifier: viewIdentifier)
-        if document == nil { print("document missing for view id \(viewIdentifier)") }
-        document?.updateAsync(update: update)
+    func newView(viewIdentifier: String, path: String?) {
+        let fileUrl = path.map() { URL(fileURLWithPath: $0) }
```

Comment:
People write "URL" differently in code. I saw that in Rust, Url is preferred while URL is the standard for Swift/Objective-C. I think having consistency between both codebases is nice, but I generally highly prefer the use of "URL" in Swift code.

Reply:
Yea, this can be annoying. As a general rule I think it's more important to be consistent with the naming conventions of a given language over being consistent between languages in a single project. You're right though that, according to the Swift [API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#general-conventions) this should be URL:

>Acronyms and initialisms that commonly appear as all upper case in American English should be uniformly up- or down-cased according to case conventions:
>
>```swift
>var utf8Bytes: [UTF8.CodeUnit]
>var isRepresentableAsASCII = true
>ar userSMTPServer: SecureSMTPServer
>```